addons/account/models/account_move.py

```diff
@@ -5401,7 +5401,7 @@ def _add_cash_basis_lines_to_exchange_difference_vals(lines, exchange_diff_move_
 
         # Fix cash basis entries.
         is_cash_basis_needed = self[0].account_internal_type in ('receivable', 'payable')
-        if is_cash_basis_needed:
+        if is_cash_basis_needed and not self._context.get('move_reverse_cancel'):
```

Comment:
If I read that in 6 months, I will wonder why we have the `not self._context.get('move_reverse_cancel')`
We should probably add a comment so our future selfs will be thankful